logsmanagement/flb_plugin.c

```diff
@@ -0,0 +1,1464 @@
+    char *docker_ev_action = NULL;
+    size_t docker_ev_action_size = 0;
+    char *docker_ev_id = NULL;
+    size_t docker_ev_id_size = 0;
+    static struct resizable_key_val_arr docker_ev_attr = {0};
+    docker_ev_attr.size = 0;
+    /* FLB_DOCKER_EV case end */
+
+    /* FLB_MQTT case */
+    char *mqtt_topic = NULL;
+    size_t mqtt_topic_size = 0;
+    static char *mqtt_message = NULL;
+    static size_t mqtt_message_size_max = 0;
+    /* FLB_MQTT case end */
+
+    size_t new_tmp_text_size = 0;
+
+    msgpack_unpacked_init(&result);
+
+    int iter = 0;
+    while (dl_msgpack_unpack_next(&result, record, size, &off) == MSGPACK_UNPACK_SUCCESS) {
+        iter++;
+        m_assert(iter == 1, "We do not expect more than one loop iteration here");
+
+        flb_time_pop_from_msgpack(&tmp_time, &result, &x);
+        
+        if(likely(x->type == MSGPACK_OBJECT_MAP && x->via.map.size != 0)){
+            msgpack_object_kv* p = x->via.map.ptr;
+            msgpack_object_kv* pend = x->via.map.ptr + x->via.map.size;
+
+            /* ================================================================ 
+             * If p_file_info == NULL, it means it is a "Forward" source, so
+             * we need to search for the associated p_file_info. This code can 
+             * be optimized further.
+             * ============================================================== */ 
+            if(p_file_info == NULL){
+                do{
+                    if(!strncmp(p->key.via.str.ptr, "stream guid", (size_t) p->key.via.str.size)){
+                        char *stream_guid = (char *) p->val.via.str.ptr;
+                        size_t stream_guid_size = p->val.via.str.size;
+                        debug_log( "stream guid:%.*s", (int) stream_guid_size, stream_guid);
+
+                        for (int i = 0; i < p_file_infos_arr->count; i++) {
+                            if(!strncmp(p_file_infos_arr->data[i]->stream_guid, stream_guid, stream_guid_size)){
+                                p_file_info = p_file_infos_arr->data[i];
+                                // debug_log( "p_file_info match found: %s type[%s]", 
+                                //                      p_file_info->stream_guid, 
+                                //                      log_src_type_t_str[p_file_info->log_type]);
+                                break;
+                            }
+                        }
+                    }
+                    ++p;
+                    // continue;
+                } while(p < pend);
+            }
+            if(unlikely(p_file_info == NULL)) goto skip_collect_and_drop_logs;
+            
+
+            uv_mutex_lock(&p_file_info->flb_tmp_buff_mut);
+            buff = p_file_info->circ_buff;
+
+
+            p = x->via.map.ptr;
+            pend = x->via.map.ptr + x->via.map.size;
+            do{
+                // if (unlikely(p->key.type != MSGPACK_OBJECT_STR || p->val.type != MSGPACK_OBJECT_STR)) {
```

Comment:
Do we really need to keep this commented code?

Reply:
Not really, it was just for testing which I have removed now!